src/System.Private.CoreLib/shared/System/Threading/Tasks/TaskContinuation.cs

```diff
@@ -699,9 +697,9 @@ protected void RunCallback(ContextCallback callback, object state, ref Task curr
                     ExecutionContext.RunInternal(context, callback, state);
                 }
             }
-            catch (Exception exc) // we explicitly do not request handling of dangerous exceptions like AVs
+            catch (Exception exception) // we explicitly do not request handling of dangerous exceptions like AVs
             {
-                ThrowAsyncIfNecessary(exc);
+                AsyncMethodBuilderCore.ThrowAsync(exception, targetContext: null);
```

Comment:
@stephentoub, in this case, there is `m_capturedContext` which is used for the callback execution. Should it also throw on it?